### ISSUE TITLE
coreml model size is much bigger than frozen graph size

### ISSUE THREAD

**Issue by ilovecv** [tf-coreml/tf-coreml#169]
Hi,

The size of the frozen graph is around 4MB. When I used tf-coreml to convert the frozen graph to coreml model, the resulting size is around 78MB, which is also larger than the checkpoint size. Do you know why? The model conversion from frozen graph to coreml model seems fine.
Thank you very much!

**Comment by aseemw** [comment]
This is not expected. 
Can you shed more details on the type of model? 
Do TF and CoreML predictions match?

**Comment by ilovecv** [comment]
Hi,
Thanks for the quick reply.
The model is an encoder-decoder style network for semantic segmentation. There are 5 denseblocks in the encoder and 5 denseblocks in the decoder. The input is an image with 256x256x3, and the output are semantic feature maps with 256x256x10. 

Here are the frozen graph and coreml model files, I also include the inspection files generated by the visualization tool in utils.

I haven't tested the coreml model yet. I will update the test results later.

[model.zip](https://github.com/tf-coreml/tf-coreml/files/1961591/model.zip)

**Comment by ilovecv** [comment]
Yes, I used the .pb file for convertion. Please see the new attachment.
I think maybe the problem is the dropout operation.
I found when I didn't use the dropout, the converted coreml model is almost the same size as the frozen model.

[coreml.zip](https://github.com/tf-coreml/tf-coreml/files/1961710/coreml.zip)

**Comment by aseemw** [comment]
Yeah, maybe the dropout mask is being converted to the CoreML model and its big. 
Since dropout is specific to training, yes the correct thing to do is to save out an inference graph without dropout and then convert the model.

**Comment by ilovecv** [comment]
Do you know if there is a good way to remove the dropout operation?
I found a related answer [validator error decoding CoreML model #168](https://github.com/tf-coreml/tf-coreml/issues/168) here. But I am not very sure how to do it. Thanks!

**Comment by aseemw** [comment]
No, I am also looking for the correct way to do it post freezing the graph!

**Comment by ilovecv** [comment]
Hi, I retrained the network without dropout layer, so I got a coreml model with the size similar as the frozen graph model. However, the output of the TF and CoreML predictions are different.

The pb, mlmodel and a test script are attached. The input image is normalized to (0, 1).
Could you please take a look at it? Thanks!

[coreml.zip](https://github.com/tf-coreml/tf-coreml/files/1962530/coreml.zip)

**Comment by aseemw** [comment]
I think the model uses instance batch normalization but the converter was sort of ignoring that. 
I have put up a fix at PR #170. Please try again with that branch.

**Comment by ilovecv** [comment]
Thank you very much for your quick reply.
I tried it and it worked!
